commit 3f77dcb272989bcc6699a96c8afe26cd4461f09a (refs/changes/76/4976/1)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-10-18T18:32:16+00:00 (12 months ago)
    
    MANTA-4004 muskie audit should check req.metadata when logging delete metrics

diff --git a/lib/audit.js b/lib/audit.js
index 795bfea..fcfee03 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -299,7 +299,7 @@ function auditLogger(options) {
             latency_histogram.observe(obj.latencyToFirstByte, labels);
         }
 
-        if (op === 'DELETE') {
+        if (op === 'DELETE' && req.metadata) {
             var md = req.metadata;
             var owner = md.creator || md.owner;
 
@@ -309,17 +309,19 @@ function auditLogger(options) {
              * the deleted object was processed by the accelerated deletion
              * pipeline.
              */
-            common.checkAccountSnaplinksEnabled(req, owner,
-                function (enabled) {
-                if (md.type === 'object' && md.contentLength > 0) {
-                    var storage = md.contentLength * md.sharks.length;
-                    labels = {
-                        accelerated_gc: !enabled,
-                        owner: owner
-                    };
-                    deleted_data_counter.add(storage, labels);
-                }
-            });
+            if (owner) {
+                common.checkAccountSnaplinksEnabled(req, owner,
+                    function (enabled) {
+                    if (md.type === 'object' && md.contentLength > 0) {
+                        var storage = md.contentLength * md.sharks.length;
+                        labels = {
+                            accelerated_gc: !enabled,
+                            owner: owner
+                        };
+                        deleted_data_counter.add(storage, labels);
+                    }
+                });
+            }
         }
 
         log.info(obj, 'handled: %d', res.statusCode);
